### ISSUE TITLE
(Android) IllegalArgumentException thrown using segment version 2.1.7-beta

### ISSUE THREAD

**Issue by AlexanderEggers** [segmentio/analytics-react-native#468]
Using the latest beta build, the segment dependency is throwing the following exception:

`java.lang.IllegalArgumentException: Could not find @ReactModule annotation in com.sovranreactnative.SovranModule`

That seems to be part of the trackDeepLinks method which is located in the AnalyticsReactNativeModule. It seems like that the used SovranModule function is missing the @ReactModule annotation.

- `analytics-react-native` version: 2.1.7-beta
- React Native version: 0.67.3
- iOS or Android or both? Android

**Comment by alanjcharles** [comment]
Hi @AlexanderEggers! hope you've been well. Would you be able to  try running `gradlew clean` in your android directory? If the issue persists would you be able to provide more information here? I am unable to replicate it, I'm afraid. Here is a payload from a  successful `2.1.7` build: 

```{
  "anonymousId": "keFjsZVWVxe3tSEqMHAfp",
  "context": {
    "app": {
      "build": "1",
      "name": "betaBug",
      "namespace": "com.betabug",
      "version": "1.0"
    },
    "device": {
      "id": "7a0daf80cb4638e7",
      "manufacturer": "unknown",
      "model": "Android SDK built for arm64",
      "name": "emulator64_arm64",
      "type": "android"
    },
    "ip": "97.118.53.141",
    "library": {
      "name": "@segment/analytics-react-native",
      "version": "2.1.7"
    },
    "locale": "en-US",
    "network": {
      "cellular": false,
      "wifi": true
    },
    "os": {
      "name": "Android",
      "version": "12"
    },
    "screen": {
      "density": 2.75,
      "height": 2208,
      "width": 1080
    },
    "timezone": "America/Denver",
    "traits": {}
  },
  "event": "Application Opened",
  "integrations": {
    "Segment.io": false
  },
  "messageId": "9QyjH5FKBtjh8jGrJuV2x",
  "originalTimestamp": "2022-03-01T18:02:07.774Z",
  "properties": {
    "build": "1",
    "from_background": false,
    "version": "1.0"
  },
  "receivedAt": "2022-03-01T18:02:14.465Z",
  "sentAt": "2022-03-01T18:02:14.280Z",
  "timestamp": "2022-03-01T18:02:07.959Z",
  "type": "track",
  "writeKey": "<removed>"
}```

**Comment by AlexanderEggers** [comment]
@alanjcharles The issue is still happening. I found a patch that is working for me - maybe that will help you to understand the problem. I can see that the SovranModule.deposite function is indeed missing the `@ReactModule` annotation. That might be even the issue here?

```
diff --git a/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt b/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt
index be54a1b..12f34b3 100644
--- a/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt
+++ b/node_modules/@segment/analytics-react-native/android/src/main/java/com/segmentanalyticsreactnative/AnalyticsReactNativeModule.kt
@@ -194,16 +194,16 @@ class AnalyticsReactNativeModule : ReactContextBaseJavaModule, ActivityEventList
   }
 
   override fun onNewIntent(intent: Intent?) {
-    Log.d(name, "onNewIntent = ${intent}")
-    trackDeepLinks(intent)
+    //Log.d(name, "onNewIntent = ${intent}")
+    //trackDeepLinks(intent)
   }
 
   override fun onHostResume() {
-    if (currentActivity != null && isColdLaunch) {
-      isColdLaunch = false
-      Log.d(name, "onHostResume = ${currentActivity!!.intent}")
-      trackDeepLinks(currentActivity!!.intent)
-    }
+    //if (currentActivity != null && isColdLaunch) {
+      //isColdLaunch = false
+      //Log.d(name, "onHostResume = ${currentActivity!!.intent}")
+      //trackDeepLinks(currentActivity!!.intent)
+    //}
   }
 
   override fun onHostPause() {
```

**Comment by alanjcharles** [comment]
thanks again @AlexanderEggers just pushed a fix in `2.1.8-beta`. Please just let me know if the issue persists.
